test/lib/mutation_source_test.cc

```diff
@@ -1343,7 +1340,7 @@ void test_slicing_with_overlapping_range_tombstones(tests::reader_concurrency_se
 
     // Check fast_forward_to()
     {
-        auto rd = ds.make_reader(s, semaphore.make_permit(), query::full_partition_range, s->full_slice(), default_priority_class(),
+        auto rd = ds.make_fragment_v1_stream(s, semaphore.make_permit(), query::full_partition_range, s->full_slice(), default_priority_class(),
```

Comment:
This test should just be migrated to v2.

Reply:
There is a `mutation.partition.apply(...)` call in there, 9 lines below, which takes `mutation_fragment`.  What's the general strategy of migrating such tests?
(Same reason goes for the other remaining non-deliberately-v1 tests, unless I missed any)

Reply:
We can add `mutation_partition::apply(mutation_fragment_v2)`. Although it will be tricky for range tombstone changes. I'm not yet sure what would be the best API for applying those. The problem is that currently the order of `mf`:s applied doesn't matter. If we converted range tombstone changes to range tombstones (as they are represented internally) based on the current + last range tombstone change, suddenly order matters.

Reply:
I was thinking that once we have the `mutation_partition` internal as v2 this problem will be gone... but no. Applying `apply(mutation_fragment_v2)` will still be order sensitive, so we just have to bite the bullet and make sure all users are aware. We can introduce an `apply_delete(position_in_partition start, position_in_partition end, tombstone)` overload so a single range delete can be easily applied (will replace the current `apply_delete(range_tombstone)`).

I'm not sure whether to defer this or not. On the one hand I would like the new v1 stream to have the smallest API possible, on the other hand I don't want to balloon this PR to huge size. You can do whichever you think is more reasonable, just explain it in the cover letter please.